hipcheck/src/plugin/mod.rs

```diff
@@ -0,0 +1,13 @@
+mod manager;
+mod types;
+
+use crate::plugin::manager::*;
+pub use crate::plugin::types::*;
+
+pub fn dummy() {
+	let plugin = Plugin {
+		name: "dummy".to_owned(),
+		entrypoint: "./dummy".to_owned(),
+	};
+	let manager = PluginExecutor::new(3, 5, 40000..u16::MAX, 1000, 10);
```

Comment:
This feels like a lot of magic numbers in a row. It would be nice to have their meanings alongside the values.
I would normally suggest using the syntax `PluginExecutor { max_spawn_attempts: 3, ...}`, but you can't directly instantiate the struct.

For a more user-facing struct, the Builder pattern would probably be a good choice. I'm not sure if it's worth it here for `PluginExecutor`, especially since it probably won't be constructed many places in the codebase.

Perhaps just some comments would be the best choice here.

Reply:
That's right, my intent is that these items become configurable in the future.